redisinsight/ui/src/pages/vector-search/manage-indexes/IndexSection.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState } from 'react'
+import { CategoryValueList, Section, SectionProps } from '@redis-ui/components'
+import { useDispatch, useSelector } from 'react-redux'
+import { CategoryValueListItem } from '@redis-ui/components/dist/Section/components/Header/components/CategoryValueList'
+import { RedisString } from 'uiSrc/slices/interfaces'
+import { bufferToString, formatLongName, stringToBuffer } from 'uiSrc/utils'
+import { sendEventTelemetry, TelemetryEvent } from 'uiSrc/telemetry'
+import { connectedInstanceSelector } from 'uiSrc/slices/instances/instances'
+import {
+  deleteRedisearchIndexAction,
+  fetchRedisearchInfoAction,
+} from 'uiSrc/slices/browser/redisearch'
+import {
+  IndexInfoDto,
+  IndexDeleteRequestBodyDto,
+} from 'apiSrc/modules/browser/redisearch/dto'
+import { IndexAttributesList, IndexInfoTableData } from './IndexAttributesList'
+
+export interface IndexSectionProps extends Omit<SectionProps, 'label'> {
+  index: RedisString
+}
+
+export const IndexSection = ({ index, ...rest }: IndexSectionProps) => {
+  const dispatch = useDispatch()
+  const indexName = bufferToString(index)
+  const { id: instanceId } = useSelector(connectedInstanceSelector)
+
+  const [tableData, setTableData] = useState<IndexInfoTableData[]>([])
+  const [indexSummaryInfo, setIndexSummaryInfo] = useState<
+    CategoryValueListItem[]
+  >(parseIndexSummaryInfo({} as IndexInfoDto))
+
+  useEffect(() => {
+    dispatch(
+      fetchRedisearchInfoAction(indexName, (data) => {
+        const indexInfo = data as unknown as IndexInfoDto
+
+        setTableData(parseIndexAttributes(indexInfo))
+        setIndexSummaryInfo(parseIndexSummaryInfo(indexInfo))
+      }),
+    )
+  }, [indexName, dispatch])
+
+  const handleDelete = () => {
+    // TODO: Replace with confirmation popup once the design is ready
+    // eslint-disable-next-line no-restricted-globals
+    const result = confirm('Are you sure you want to delete this index?')
+
+    if (result) {
+      const data: IndexDeleteRequestBodyDto = {
+        index: stringToBuffer(indexName),
+      }
+
+      dispatch(deleteRedisearchIndexAction(data, onDeletedIndexSuccess))
+    }
+  }
+
+  const onDeletedIndexSuccess = (data: IndexDeleteRequestBodyDto) => {
+    sendEventTelemetry({
+      event: TelemetryEvent.SEARCH_INDEX_DELETED,
+      eventData: {
+        databaseId: instanceId,
+        indexName: data.index,
+      },
+    })
+  }
+
+  return (
+    <Section
+      collapsible
+      collapsedInfo={<CategoryValueList categoryValueList={indexSummaryInfo} />}
+      content={<IndexAttributesList data={tableData} />}
+      // TODO: Add FieldTag component to list the types of the different fields
+      label={formatLongName(indexName)}
```

Comment:
Currently, the `Section` component ([docs](https://redislabsdev.github.io/redis-ui/?path=/docs/components-section--docs)) doesn't support anything other than a string for its label, so we can't make it look like the design.

<img width="1222" height="252" alt="image" src="https://github.com/user-attachments/assets/2b3820d8-f41a-4d40-a4d1-7d33bdb967b6" />